hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotCache.java

```diff
@@ -231,23 +216,20 @@ public ReferenceCounted<IOmMetadataReader, SnapshotCache> get(String key,
    * @param key snapshot table key
    */
   public void release(String key) {
-    ReferenceCounted<IOmMetadataReader, SnapshotCache>
-        rcOmSnapshot = dbMap.get(key);
-    if (rcOmSnapshot == null) {
-      throw new IllegalArgumentException(
-          "Key '" + key + "' does not exist in cache");
-    }
-
-    if (rcOmSnapshot.decrementRefCount() == 0L) {
-      synchronized (pendingEvictionList) {
-        // v is eligible to be evicted and closed
-        pendingEvictionList.add(rcOmSnapshot);
+    Lock lock = getLock(key);
+    lock.lock();
+    try {
+      ReferenceCounted<IOmMetadataReader, SnapshotCache> rcOmSnapshot = dbMap.get(key);
+      if (rcOmSnapshot == null) {
+        throw new IllegalArgumentException("Key '" + key + "' does not exist in cache");
       }
+      rcOmSnapshot.decrementRefCount();
+      // The cache size might have already exceeded the soft limit
+      // Thus triggering cleanup() to check and evict if applicable
+      cleanup();
```

Comment:
Though this is not used currently, We can move this as well.

Reply:
I missed :face_palm:. Thanks for the catch. Removed.